### PR TITLE
Revised ASP.NET Core module installer for Windows ARM64

### LINKED ISSUE #47115: ASP.NET Core module installer should be improved further for Windows ARM64
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Is your feature request related to a problem? Please describe the problem.

The current installer (.NET 7 for example) does the following on Windows 11 ARM64,

1. Install pure ARM64 build of the files to `Program Files`.
1. Install pure ARM64 build of the files to `Program Files (x86)`.

So, if people try to create three application pools on IIS for ARM64, x64, and x86, they can only get things running in the ARM64 pool (`enableEmulationOnWinArm64` set to `false` and `enable32BitAppOnWin64` set to `false`). All other pools will crash.

### Describe the solution you'd like

The changes should be made are,

1. Install x86 build to `Program Files (x86)`, which resolves x86 application pools.
1. Compile pure forwarders `aspnetcorev2.dll` and `aspnetcorev2_outofprocess.dll` and install them along with arm64/x64 bits to `Program Files`. This resolves both in-process and out-of-process modes for arm64/x64 application pools.

> Note that an alternative way is to compile ARM64X builds of both `aspnetcorev2.dll` and `aspnetcorev2_outofprocess.dll` and install to `Program Files`. However, it raised many challenges and needs further investigation.

This should allow all three kinds of application pools to run properly and maximize compatibility.

A all-in-one pull request is currently opened, #47290.

> Note that #47124 is no longer needed.

### Additional context

I tried to build ASP.NET Core module with a ARM64X profile, but found quite a few hurdles (changes to compiler settings, extra defines and missing files to include). I guess that's why this hasn't been finished in .NET 7 timeline.

Hope it can be done in .NET 8 to complete Windows ARM64 support.

### PR DESCRIPTION
**Pull request by lextm**
# Revised ASP.NET Core module installer for Windows ARM64

- [x] You've read the [Contributor Guide](https://github.com/dotnet/aspnetcore/blob/main/CONTRIBUTING.md) and [Code of Conduct](https://github.com/dotnet/aspnetcore/blob/main/CODE-OF-CONDUCT.md).
- [ ] You've included unit or integration tests for your change, where applicable.
- [ ] You've included inline docs for your change, where applicable.
- [x] There's an open issue for the PR that you are making. If you'd like to propose a new feature or change, please open an issue to discuss the change or find an existing issue.

* Switch to `LoadLibraryEx` with `LOAD_LIBRARY_SEARCH_DLL_LOAD_DIR` flag set, so that the pure forwarder works as desired.
* Add ARM64X pure forwarders.
* Revise WiX installer to utilize ARM64X pure forwarders

Fixes #47115.

### About the change in `HandlerResolver.cpp`
To implement all proposed installer changes in #47115, then in the out-of-process folder three files present,

1. `aspnetcorev2_outofprocess.dll`, an ARM64X pure forwarder.
1. `aspnetcorev2_outofprocess_arm64.dll`, the pure ARM64 handler.
1. `aspnetcorev2_outofprocess_x64.dll`, the x64 handler.

Before this change, `LoadLibrary` only loads the pure forwarder, but then scans the default folders (the dll folder is not included) to load the actual handler which obviously fails.

After this change, `LoadLibraryEx` can pick up the correct handler as the dll folder is added in the search folder list.

### About files in `Forwarders` folder
Source code for the two ARM64X pure forwarders.

### About changes in `ANCMV2` folder
Before this change, only ARM64 web apps can run on IIS.

After this change, the new file structure enables ARM64/x64/x86 side by side execution on IIS.

### CODE REVIEW

File: src/Installers/Windows/AspNetCoreModule-Setup/ANCMV2/aspnetcoremodulev2.wxs [hunk 1]
```diff
@@ -309,6 +343,7 @@
             </Row>
         </CustomTable>
 
+        <!-- <?if $(var.Platform) = "x86" OR $(var.Platform) = "x64" ?> -->
```
Comment (lextm) [inline src/Installers/Windows/AspNetCoreModule-Setup/ANCMV2/aspnetcoremodulev2.wxs:346]: For some unknown reason, this section has to be commented out.

**Comment by wtgodbe** [comment]
Looks like we're hitting some errors in the new Forwarder project, are you seeing these locally?

> The system cannot find the path specified.
  'cl' is not recognized as an internal or external command,
  operable program or batch file.
  The system cannot find the path specified.
  'cl' is not recognized as an internal or external command,
  operable program or batch file.
  The system cannot find the path specified.
  A subdirectory or file D:\a\_work\1\s\artifacts\obj\\AspNetCoreModuleForwarders already exists.

**Comment by lextm** [comment]
I don't see the errors locally, but expected something like this. I will dig into the CI log files and see what further changes are needed.

**Comment by lextm** [comment]
I was able to reproduce the `cl` errors by running `.\eng\build.cmd -ci xxxx` as it seems to set up the build environment differently from `build.ps1` under `src\Installers\Windows`.

However, after applying the changes CI still failed and this time I am not familiar with the errors.

**Comment by wtgodbe** [comment]
> However, after applying the changes CI still failed and this time I am not familiar with the errors.

That looks like a transient error, I hit rerun.

To be transparent, we're still trying to figure out if we want to take this change. We think the number of customers who use ANCM/IIS on win-arm64 is relatively low, and we may not have the resources on our team to provide support for this new scenario. If you could share some info on how strong your need is for this feature, that would help us come to a decision.

**Comment by mkArtakMSFT** [comment]
Closing as we don't plan to take this change at this time.

**Pull request closed without merging**